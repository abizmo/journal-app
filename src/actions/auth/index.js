export { default as login } from './login';
export { default as loginWithEmail } from './loginWithEmail';
export { default as loginWithGoogle } from './loginWithGoogle';
export { default as logout } from './logout';
export { default as signOut } from './signOut';
export { default as signUp } from './signUp';

export const LOGIN = 'LOGIN';
export const LOGOUT = 'LOGOUT';

export const types = {
  LOGIN,
  LOGOUT,
};
