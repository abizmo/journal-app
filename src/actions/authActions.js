import Swal from "sweetalert2";
import {
  loginWithGoogleProvider,
  signInWithEmail,
  signOutInFirebase,
  signUpWithEmailAndPassword
} from "../firebase";
import { startAction, stopAction } from "./uiActions";

export const LOGIN = 'LOGIN';
export const LOGOUT = 'LOGOUT';

export const login = (user) => ({
  type: LOGIN,
  payload: user,
});

export const logout = () => ({
  type: LOGOUT,
});

export const loginWithEmail = ({ email, password }) => (dispatch) => {
  dispatch(startAction());
  signInWithEmail({ email, password })
    .then((user) => {
      dispatch(stopAction());
      dispatch(login(user));
    })
    .catch(({ message }) => {
      dispatch(stopAction());
      Swal.fire('Error', message, 'error');
    });
};

export const loginWithGoogle = () => (dispatch) => {
  loginWithGoogleProvider()
    .then((user) => dispatch(login(user)))
    .catch(({ message }) => Swal.fire('Error', message, 'error'));
};

export const signUp = (newUser) => (dispatch) => {
  signUpWithEmailAndPassword(newUser)
    .then((user) => dispatch(login(user)))
    .catch(({ message }) => Swal.fire('Error', message, 'error'));
};

export const signOut = () => (dispatch) => {
  signOutInFirebase()
    .then(() => dispatch(logout()))
    .catch(({ message }) => Swal.fire('Error', message, 'error'));
};
