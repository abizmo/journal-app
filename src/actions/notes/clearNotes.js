import { CLEAR_NOTES } from ".";

const clearNotes = () => ({
  type: CLEAR_NOTES,
});

export default clearNotes;
