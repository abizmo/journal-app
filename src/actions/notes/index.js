export { default as setActive } from './setActive';
export { default as refreshNote } from './refreshNote';
export { default as clearNotes } from './clearNotes';
export { default as getNotes } from './getNotes';
export { default as uploadImage } from './uploadImage';
export { default as removeNote } from './removeNote';
export { default as newNote } from './newNote';
export { default as saveNote } from './saveNote';

export const ADD_NOTE = 'ADD_NOTE';
export const CLEAR_NOTES = 'CLEAR_NOTES'
export const DELETE_NOTE = 'DELETE_NOTE';
export const SET_ACTIVE = 'SET_ACTIVE';
export const SET_NOTES = 'SET_NOTES';
export const UPDATE_NOTE = 'UPDATE_NOTE';

export const types = {
  ADD_NOTE,
  CLEAR_NOTES,
  DELETE_NOTE,
  SET_ACTIVE,
  SET_NOTES,
  UPDATE_NOTE,
};
