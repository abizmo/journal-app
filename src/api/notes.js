import {
  addDoc,
  collection,
  doc,
  getDocs,
  updateDoc,
} from "firebase/firestore";
import { db } from "../firebase";

export const createNote = (userId, note) => addDoc(collection(db, `${userId}/journal/notes/`), note)
  .then(({ id }) => id);

export const getAllNotes = (userId) => getDocs(collection(db, `${userId}/journal/notes/`))
  .then((res) => {
    const notes = [];
    res.forEach((doc) => {
      notes.push({ ...doc.data(), id: doc.id });
    });
    
    return notes;
});

export const updateNote = (userId, noteId, data) => {
  const docRef = doc(db, `${userId}/journal/notes/${noteId}`);

  return updateDoc(docRef, data)
    .then(() => data);
};