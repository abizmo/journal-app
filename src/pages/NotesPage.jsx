import React from 'react'
import NotesBar from '../styles/components/NotesBar'

const NotesPage = () => {
  return (
    <div className="notes__container">
      <NotesBar />
      <p>notes</p>
    </div>
  )
}

export default NotesPage
