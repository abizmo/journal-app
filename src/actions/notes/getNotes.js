import { SET_NOTES } from ".";
import { getAllNotes } from "../../api/notes";
import { error } from "../../helpers/alert";

const setNotes = (notes) => ({
  type: SET_NOTES,
  payload: notes,
});

const getNotes = () => (dispatch, getState) => {
  const { userId } = getState().auth;

  getAllNotes(userId)
    .then((notes) => {
      dispatch(setNotes(notes));
    })
    .catch(({ message }) => error(message));
};

export default getNotes;
