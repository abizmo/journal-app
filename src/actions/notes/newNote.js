import { ADD_NOTE } from ".";
import { createNote } from "../../api/notes";
import { error } from "../../helpers/alert";

const addNote = (note) => ({
  type: ADD_NOTE,
  payload: note,
});

const newNote = () => (dispatch, getState) => {
  const { userId } = getState().auth;

  const note = {
    title: '',
    body: '',
    date: new Date().getTime(),
  };

  createNote(userId, note)
    .then((id) => {
      dispatch(addNote({ ...note, id }))
    })
    .catch(({ message }) => error(message));
};

export default newNote;
