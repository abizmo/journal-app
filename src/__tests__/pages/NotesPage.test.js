import { mount } from "enzyme";
import { Provider } from "react-redux";
import createMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import NotesPage from "../../pages/NotesPage";
import { notes } from '../../fixtures/notes'
import { setActive } from "../../actions/notes";
jest.mock('../../actions/notes');

const initialState = {
  notes: {
    active: {
      ...notes[0],
    },
    notes,
  }
};
const middlewares = [thunk];
const mockStore = createMockStore(middlewares);

describe('<NotesPage />', () => {
  const store = mockStore(initialState);
  store.dispatch = jest.fn();
  const wrapper = mount(
    <Provider store={store}>
      <NotesPage />
    </Provider>
  );

  test('should render and match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  
  test('should dispatch setActive', () => {
    wrapper.find('input[name="title"]').simulate('change', {
      target: {
        name: 'title',
        value: 'Test title',
      },
    });

    expect(setActive).toHaveBeenCalledWith({ ...notes[0], title: 'Test title' });
  });
});
