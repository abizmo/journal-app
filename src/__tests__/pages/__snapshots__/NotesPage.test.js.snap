// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotesPage /> should render and match snapshot 1`] = `
<div
  className="notes__container"
>
  <div
    className="notes__bar-container"
  >
    <p>
      19th March 2021
    </p>
    <input
      id="fileInput"
      name="fileInput"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
    <div>
      <button
        className="btn btn-outline-white mr-1"
        onClick={[Function]}
      >
        Add Picture
      </button>
      <button
        className="btn btn-white"
        onClick={[Function]}
      >
        Save
      </button>
    </div>
  </div>
  <div
    className="notes__new-note"
  >
    <input
      autoComplete="off"
      className="notes__input"
      name="title"
      onChange={[Function]}
      placeholder="An awesome note"
      type="text"
      value="Titulo 1"
    />
    <textarea
      className="notes__textarea"
      name="body"
      onChange={[Function]}
      placeholder="Once upon a time..."
      value="Body 1"
    />
    <img
      alt="new note"
      className="notes__image"
      src="http://localhost/images/1.jpeg"
    />
  </div>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;
